esmvalcore/preprocessor/_time.py

```diff
@@ -459,6 +459,52 @@ def anomalies(cube, period):
     return cube
 
 
+def standardize(cube, period='full'):
+    """Standardize the input data with the specified granularity.
+
+    This function standardizes the input data. It calculates the anomalies
+    (x-mean(x)) and divides them through the standard deviation where both
+    anomalies and standard deviation are calculated over the specified period.
+
+    Parameters
+    ----------
+    cube: iris.cube.Cube
+        input cube.
+
+    period: str, optional
+        Period for computing mean and standard deviation
+        Available periods: 'full', 'season', 'seasonal', 'monthly', 'month',
+        'mon', 'daily', 'day'
+
+    Returns
+    -------
+    iris.cube.Cube
+        Standardized cube
+
+
+    See Also
+    --------
+        anomalies
+        climate_statistics
+    """
+    cube_anomalies = anomalies(cube, period)
+    cube_stddev = climate_statistics(cube, operator='std_dev', period=period)
+
+    if period == 'full':
+        cube.data = cube_anomalies.data / cube_stddev.data
+    elif period in ['season', 'seasonal', 'monthly',
+                    'month', 'mon', 'daily', 'day']:
+        ratio = [i / j for i, j in
+                 zip(cube_anomalies.shape, cube_stddev.shape)]
```

Comment:
this will fail if `cube_anomalies.shape == cube_stddev.shape`, can you put a check and raise an exception pls?

Reply:
why exactly? I don't see why this would fail. I think it would fail only if `cube_anomalies.ndim != cube_stddev.ndim` is that the check that you would like to see?

Reply:
sorry, fail *silently* (I ate the most important word) ie zip will work fine but if the dims are not equal, it will return truncated results that will not be detectable until probably at the last func line, if at all, and even there when you are tiling, it may tile on a wrong tile grid; statistically it's important to have equal dimensionality and any silent fail will introduce a cloaked bug

Reply:
aka :dagger:

Reply:
still not clear to me, can you check what you wrote originally (3h ago)? I see your point that an extra check here is needed, but I don't think you wrote originally (`shape`) what you intended (`dims`?) to write ?

Reply:
sorry Bas, this fell swiftly though the cracks. Yes, so what I meant was if `cube_anomalies.shape != cube_stddev.shape` but the case to be reckoned with is if the dims are different (could they be?) ie `cube_anomalies.shape = (3, 4, 5)` and `cube_stddev.shape = (3, 4)` - in that case the ratios will be computed only for the common pairs `(i, j)` that `zip` returns

Reply:
Ok, now I see! Didn't expect the zipping to work when unequal lengths are passed, good to know ! I will built in a check before.